server/lib/scheduler/Assignment.ts

```diff
@@ -0,0 +1,21 @@
+class Assignment {
+  private startedAt;
+  private userId;
+  private workspaceId;
+
+  public constructor({ userId, workspaceId, startAtTimestamp = Date.now() }){
+    this.startAtTimestamp = startAtTimestamp;
+    this.userId = userId;
+    this.workspaceId = workspaceId;
+  }
+
+  public getStartedAtTimestamp() {
+    return this.startAtTimestamp;
+  }
+
+  public getWorkspaceId() {
+    return this.workspaceId;
+  }
+}
```

Comment:
Why did you make this its' own class? If you leave it as an object it actually may be a bit more flexible for querying and so on.

Reply:
If you did so for the Date.now() in the constructor, you could still have a contructor-like function.

Reply:
It just made sense conceptually to me to have an assignment as a stand-alone entity. You could imagine a system in the future that passed assignments around. This is also why I gave them userId's even though this isn't needed the way things are currently set up.